Add simple installation/run tests with GitHub actions

koldunovn opened the pull request:
Two versions of the tests are added. Description and discussion in #422

mandresm committed:
Merge pull request #412 from esm-tools/docs/file_movements

Docs/file movements

koldunovn committed:
Create esm_tools_actions.yml

koldunovn committed:
Update esm_tools_actions.yml

koldunovn committed:
try to change the hostname for the container

koldunovn committed:
another try

koldunovn committed:
add clean run

koldunovn committed:
Create esm_tools_actions_novm.yml

koldunovn committed:
Update esm_tools_actions_novm.yml

koldunovn committed:
remove 2.7

koldunovn committed:
add hostname

koldunovn committed:
fighting with hostname change

koldunovn committed:
spelling

koldunovn committed:
add 3.8 and 3.9

pgierz commented on .github/workflows/esm_tools_actions.yml:
```diff
@@ -0,0 +1,63 @@
+      run: |    
+        esm_plugins
+    
+    - name: run esm_plugins
```
We don't need that one twice

koldunovn replied:
Oh, fixed.

pgierz commented on .github/workflows/esm_tools_actions_novm.yml:
```diff
@@ -0,0 +1,66 @@
+          LANG: en_US.UTF-8
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.7', '3.8', '3.9']
```
I think we still have at least one machine using Python 3.5. But we also use f-strings, which aren't in there until 3.6....so, that will fly around the ears at some point.

koldunovn replied:
I added 3.5, and it failed with `SyntaxError: invalid syntax` (https://github.com/esm-tools/esm_tools/runs/3242808547?check_suite_focus=true). I will remove now. It was actually a good test showing it's working :)

pgierz replied:
Yup, we get a big red X. I'll adapt the setup.py to ensure it doesn't even try to install itself if you aren't using 3.6 or newer. 

I at some point for one my private projects also had figured out how to get the CI to automatically publish stuff to PyPI. This might also be something to look into

koldunovn replied:
@pgierz looks like there is an action for this :) https://github.com/pypa/gh-action-pypi-publish One could probably do it for tags or stable releases, though :)

pgierz reviewed (commented):
Hi Nikolay, thanks very very VERY much for putting this together. I will be a great starting place.

koldunovn committed:
Remove esm_plugins double call

koldunovn committed:
add 3.5 and see if it will fail :)

pgierz committed:
Update setup.py

This one will still fail, since the CI will try to install itself with python 3.5. But it should now fail with a "I am not allowed to install on python 3.5". If yes, then I'll update the CI to throw that one out again.

koldunovn committed:
remove 3.5

pgierz commented:
....hooray we are slowly moving into the modern world 🥳

koldunovn committed:
try to change hostname only once for the step

mandresm commented:
Thanks a lot Nikolay! :) This is awesome!

mandresm reviewed (approved):
I think we can merge this into `develop` and `prep_release` already, so that we have this very basic CI already in place, and we can expand it soon through other PRs.
